### PR TITLE
Fix singel select filter control issue

### PR DESCRIPTION
**Pull request by Falseee**
fix for #1782 
Fixed JsFiddle : https://jsfiddle.net/esnn4trv/7/

And added new table option to maintain selected rows on filter

### CODE REVIEW

**Comment by djhvscf** [comment]
Your example is not working for me @Falseee

**Comment by Falseee** [comment]
@djhvscf Its working for me, if i check a row and search for something and remove the search no checkbox is checked. (so you cant check 2 checkboxes when singel select is true)

**Comment by djhvscf** [comment]
Yes, but when I clean the search input the checkbox is not checked.. Did I miss something?

File: src/extensions/filter-control/bootstrap-table-filter-control.js [hunk 1]
```diff
@@ -292,8 +294,25 @@
...
+        if (singleSelect) {
+            $('tbody tr td.bs-checkbox input:checked').click();
+        }
+
+        if(!that.options.filterMaintainSelected){
+            var checkedCheckboxes = $('tbody tr td.bs-checkbox input[type="checkbox"]:checked');
+            $.each(checkedCheckboxes, function (i, item) {
+                item.click();
+            });
+            console.log(that);
```
Comment (djhvscf) [inline src/extensions/filter-control/bootstrap-table-filter-control.js]: Delete this line, please

**Comment by Falseee** [comment]
@djhvscf No you missed nothing, if you search the checkbox will be unchecked permanent.
If this solution is not good enough close this pull request, otherwise i remove the debug log(console.log) :)

**Comment by djhvscf** [comment]
I think if I checked something it still checked after any action in the table

**Comment by Falseee** [comment]
Okay, then please close this PR.
My knowledge about this code is not good enough to fix it on this way.

**Pull request closed without merging**